### PR TITLE
v1.10: usnic fixes

### PR DESCRIPTION
**Pull request by jsquyres**
- ensure all messages are sent on the data channel
- fix wraparound sequence number issue
- don't overrun fi_av_insert() EQ
- better handling for AV EQ length default value

@bturrubiates Please review

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1287/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1290/ for details.

File: ompi/mca/btl/usnic/btl_usnic_module.c [hunk 1]
```diff
@@ -1205,7 +1269,7 @@ usnic_send(
         /* assign length */
         sseg->ss_len = sizeof(opal_btl_usnic_btl_header_t) + frag->sf_size;
 
-        sseg->ss_channel = USNIC_PRIORITY_CHANNEL;
+        sseg->ss_channel = USNIC_DATA_CHANNEL;
```
Comment (bturrubiates) [inline ompi/mca/btl/usnic/btl_usnic_module.c:1272]: So before this change, all sends were going over the priority channel (is that for acks?)

Comment (jsquyres) [inline ompi/mca/btl/usnic/btl_usnic_module.c:1272]: Just small sends.  See old line 1178 (a few lines above this one):

``` c
if (frag->sf_base.uf_type == OPAL_BTL_USNIC_FRAG_SMALL_SEND &&
             frag->sf_ack_bytes_left < module->max_tiny_payload &&
// ...etc.
```

**Comment by bturrubiates** [comment]
:+1:

**Comment by jsquyres** [comment]
Ben gave a :+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1291/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1294/ for details.

**Comment by jsquyres** [comment]
Pushed one additional commit as a result from @bturrubiates' review.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1296/ for details.

**Comment by rhc54** [comment]
@jsquyres any urgency to this?

**Comment by jsquyres** [comment]
A release "sometime soon" would be nice -- I don't need an immediate release, however.  Similar to other vendors, it's enough to know that it's actually merged in the community upstream.

**Pull request merged**